app/javascript/packs/commentDropdowns.js

```diff
@@ -15,51 +15,99 @@ const handleCopyPermalink = (closeDropdown) => {
 };
 
 const initializeArticlePageDropdowns = () => {
+  // Gather all dropdown triggers for comment options and profile previews
   const dropdownTriggers = document.querySelectorAll(
-    'button[id^=comment-dropdown-trigger]',
+    'button[id^=comment-dropdown-trigger], button[id^=comment-profile-preview-trigger-]',
   );
+
   for (const dropdownTrigger of dropdownTriggers) {
     if (dropdownTrigger.dataset.initialized) {
       //  Make sure we only initialize once
       continue;
     }
 
     const dropdownContentId = dropdownTrigger.getAttribute('aria-controls');
+    const dropdownElement = document.getElementById(dropdownContentId);
 
-    const { closeDropdown } = initializeDropdown({
-      triggerElementId: dropdownTrigger.id,
-      dropdownContentId,
-    });
+    if (dropdownElement) {
+      const { closeDropdown } = initializeDropdown({
+        triggerElementId: dropdownTrigger.id,
+        dropdownContentId,
+      });
 
-    // Add actual link location (SEO doesn't like these "useless" links, so adding in here instead of in HTML)
-    const dropdownElement = document.getElementById(dropdownContentId);
-    const reportAbuseWrapper = dropdownElement.querySelector(
-      '.report-abuse-link-wrapper',
-    );
-    if (reportAbuseWrapper) {
-      reportAbuseWrapper.innerHTML = `<a href="${reportAbuseWrapper.dataset.path}" class="crayons-link crayons-link--block">Report abuse</a>`;
-    }
+      // Add actual link location (SEO doesn't like these "useless" links, so adding in here instead of in HTML)
+      const reportAbuseWrapper = dropdownElement.querySelector(
+        '.report-abuse-link-wrapper',
+      );
+      if (reportAbuseWrapper) {
+        reportAbuseWrapper.innerHTML = `<a href="${reportAbuseWrapper.dataset.path}" class="crayons-link crayons-link--block">Report abuse</a>`;
+      }
 
-    // Initialize the "Copy link" functionality
-    dropdownElement
-      .querySelector('.permalink-copybtn')
-      ?.addEventListener('click', handleCopyPermalink(closeDropdown));
+      // Initialize the "Copy link" functionality
+      dropdownElement
+        .querySelector('.permalink-copybtn')
+        ?.addEventListener('click', handleCopyPermalink(closeDropdown));
 
-    dropdownTrigger.dataset.initialized = 'true';
+      dropdownTrigger.dataset.initialized = 'true';
+    }
   }
 };
 
+/**
+ * Function to retrieve the Profile Preview Card for newly added comments, and asynchronously replace the placeholder
+ *
+ * @param {HTMLElement} placeholderElement The <span> placeholder element to be replaced by the preview dropdown
+ */
+const fetchMissingProfilePreviewCard = async (placeholderElement) => {
+  const htmlContent = await window
+    .fetch(
+      `/profile_preview_card/show?userid=${placeholderElement.dataset.jsCommentUserId}&preview_card_id=${placeholderElement.dataset.jsDropdownContentId}`,
+    )
+    .then((res) => res.text());
```

Comment:
Interesting combination of using both `await` and `then`. I tend to use the pseudo imperative style:

```js
const response = await fetch
const text = await response.text()
```

TIL!

Reply:
Yeah I tend to keep everything related to the fetch (including extracting the relevant response) to the one await 🤷‍♀️  Now that I'm thinking about it I'm not sure if that's weird 😄

Reply:
Everything is Promises which is why you can do that with async/await and `.then`.

Reply:
Nitpick (non-blocking): Everything is Promises which is why you can do that with async/await and `.then`. Having said that, I would opt for await all the way, but this is a preference.

Reply:
FWIW, I'm with Nick on this one, i.e. either promises all the way or "imperative" `await` all the way, mixing the 2 styles is a bit confusing.